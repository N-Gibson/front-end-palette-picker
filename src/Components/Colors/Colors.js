import React from 'react';
import '../Colors/Colors.scss';
import lockIcon from '../../images/locked-icon.png'

export const Colors = (props) => {
  const {color1, color2, color3, color4, color5, lockColor} = props;

  const lockedStyle = {
    opacity: 1
  }

  return(
    <section className="colors-page">
      <div 
        className="color div__color--1" 
        style={{backgroundColor: color1.hex}}>
          {!color1.isLocked && <img 
            alt="lock icon" 
            src={lockIcon} 
            onClick={() => lockColor(1)}/>}
          {color1.isLocked  && <img 
            alt="lock icon" 
            style={lockedStyle} 
            src={lockIcon} 
            onClick={() => lockColor(1)}/>}
        <h2 className="color__hex">{color1.hex}</h2>
     </div>
     <div 
      className="color div__color--2" 
      style={{backgroundColor: color2.hex}} >
       {!color2.isLocked && <img 
          alt="lock icon" 
          src={lockIcon} 
          onClick={() => lockColor(2)}/>}
       {color2.isLocked  && <img 
          alt="lock icon" 
          style={lockedStyle} 
          src={lockIcon} 
          onClick={() => lockColor(2)}/>}
       <h2 className="color__hex">{color2.hex}</h2>
     </div>
     <div 
      className="color div__color--3" 
      style={{backgroundColor: color3.hex}}>
       {!color3.isLocked && <img 
         alt="lock icon" 
         src={lockIcon} 
         onClick={() => lockColor(3)}/>}
       {color3.isLocked &&  <img
         alt="lock icon"
         style={lockedStyle}
         src={lockIcon}
         onClick={() => lockColor(3)}/>}
       <h2 className="color__hex">{color3.hex}</h2>
     </div>
     <div 
      className="color div__color--4" 
      style={{backgroundColor: color4.hex}}>
       {!color4.isLocked && <img 
         alt="lock icon" 
         src={lockIcon} 
         onClick={() => lockColor(4)}/>}
       {color4.isLocked &&  <img
         alt="lock icon"
         style={lockedStyle}
         src={lockIcon}
         onClick={() => lockColor(4)}/>}
       <h2 className="color__hex">{color4.hex}</h2>
     </div>
      <div 
        className="color div__color5" 
        style={{backgroundColor: color5.hex}}>
      {!color5.isLocked && <img 
        alt="lock icon" 
        src={lockIcon} 
        onClick={() => lockColor(5)}/>}
      {color5.isLocked &&  <img
        alt="lock icon"
        style={lockedStyle}
        src={lockIcon}
        onClick={() => lockColor(5)}/>}
      <h2 className="color__hex">{color5.hex}</h2>
      </div>
    </section>
  )
}

export default Colors