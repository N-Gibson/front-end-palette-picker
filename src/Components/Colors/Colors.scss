
.colors-page {
  display: flex;
}

.color {
  border: 1px solid red;
  width: 20%;
  height: 60vh;
  margin: 0;

}