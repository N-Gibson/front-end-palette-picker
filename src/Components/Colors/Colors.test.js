import React from 'react';
import { shallow } from 'enzyme';
import Colors from './Colors'

describe('Colors', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Colors props={{color1:{hex:'1'}, color2:{hex:'2'}, color3: {hex:'3'}, color4: {hex: '4'}, color5: {hex: '5'}} />);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })
})