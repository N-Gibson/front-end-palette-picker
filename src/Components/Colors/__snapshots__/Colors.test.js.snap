// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Colors should match the snapshot 1`] = `
<section
  className="colors-page"
>
  <div
    className="color div__color--1"
    style={
      Object {
        "backgroundColor": "2",
      }
    }
  >
    <img
      alt="lock icon"
      onClick={[Function]}
      src="locked-icon.png"
    />
    <h2
      className="color__hex"
    >
      2
    </h2>
  </div>
  <div
    className="color div__color--2"
    style={
      Object {
        "backgroundColor": "2",
      }
    }
  >
    <img
      alt="lock icon"
      src="locked-icon.png"
    />
    <h2
      className="color__hex"
    >
      2
    </h2>
  </div>
  <div
    className="color div__color--3"
    style={
      Object {
        "backgroundColor": "2",
      }
    }
  >
    <img
      alt="lock icon"
      src="locked-icon.png"
    />
    <h2
      className="color__hex"
    >
      2
    </h2>
  </div>
  <div
    className="color div__color--4"
    style={
      Object {
        "backgroundColor": "2",
      }
    }
  >
    <img
      alt="lock icon"
      src="locked-icon.png"
    />
    <h2
      className="color__hex"
    >
      2
    </h2>
  </div>
  <div
    className="color div__color5"
    style={
      Object {
        "backgroundColor": "2",
      }
    }
  >
    <img
      alt="lock icon"
      src="locked-icon.png"
    />
    <h2
      className="color__hex"
    >
      2
    </h2>
  </div>
</section>
`;
